conan/internal/cache/home_paths.py

```diff
@@ -11,6 +11,19 @@ class HomePaths:
     """
     def __init__(self, home_folder):
         self._home = home_folder
+        self._extensions_folder = os.path.join(self._home, _EXTENSIONS_FOLDER)
+        self._plugins_folder = os.path.join(self._extensions_folder, _PLUGINS)
+        # IMPORTANT: Developer mode is not meant to be used in production environments!!
+        self._check_developer_mode()
+
+    def _check_developer_mode(self):
+        """
+        Developer mode reads several environment variables which defines the extensions/plugins
+        folders. That facilitates the debugging process if the extensions code is in another place.
+        """
+        if os.getenv("_CONAN_DEVELOPER_MODE", False):
+            self._extensions_folder = os.getenv("_CONAN_EXTENSIONS_FOLDER", self._extensions_folder)
+            self._plugins_folder = os.getenv("_CONAN_PLUGINS_FOLDER", os.path.join(self._extensions_folder, _PLUGINS))
```

Comment:
I am not convinced of:
- using env-vars for the values of folders to use extensions or plugins
- I'd limit things to just custom commands initially
- Custom commands would need to be appended, not replacing other custom commands

So ``confs`` might be better for this, lets discuss it.